airbyte-ci/connectors/metadata_service/lib/tests/test_gcs_upload.py

```diff
@@ -21,8 +22,9 @@
 VALID_DOC_FILE_PATH = Path(DOCS_PATH) / MOCK_DOC_URL_PATH
 
 
-def stub_is_image_on_docker_hub(image_name: str, version: str) -> bool:
-    return "exists" in image_name and version not in MOCK_VERSIONS_THAT_DO_NOT_EXIST
+def stub_is_image_on_docker_hub(image_name: str, version: str, digest: Optional[str] = None, retries: int = 0, wait_sec: int = 30) -> bool:
+    image_exists = all(["exists" in image_name, version not in MOCK_VERSIONS_THAT_DO_NOT_EXIST, digest is None or "exists" in digest])
```

Comment:
Can we make the "does not exist" case use `0.0.0` version.
It might be more robust. I can easily make a typo like `doesnotexists`

Reply:
I'll look into a better does not exist version number combo (i think we still need one specific to breaking changes) before merging this. 

on the "exists" - I agree. Mentioned a suggestion [here](https://github.com/airbytehq/airbyte/pull/33127#discussion_r1416499671), but will put that off to a separate pr